### PR TITLE
[CUMULUS-2451] Add logging to data-migration2 lambda and return summary object

### PR DESCRIPTION
**Pull request by botanical**
**Summary:** Summary of changes

Addresses [CUMULUS-2451: Add logging to data-migration2 lambda and return summary object](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2451)

## Changes

* Added summary type file `packages/db/src/types/summary.ts` with `MigrationSummary` and `DataMigration1` and `DataMigration2` types
* Updated `data-migration1` and `data-migration2` lambdas to return summary objects
* Added intermittent logging based on batch of 100 records processed for executions, granules and files, and PDRs
* Remove RecordAlreadyMigrated logs in `data-migration1` and `data-migration2`

## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [x] Adhoc testing
- [ ] Integration tests

### CODE REVIEW

**Comment by botanical** [comment]
I removed the logs for `RecordAlreadyMigrated` and deployed the lambdas to test that the logs do not show that error. I maintained the error throwing to maintain the skip count.

File: lambdas/data-migration2/src/executions.ts
Comment (markdboyd) [inline lambdas/data-migration2/src/executions.ts:24]: It still does throw this error?

File: lambdas/data-migration2/src/executions.ts
Comment (markdboyd) [inline lambdas/data-migration2/src/executions.ts:91]: What do you think about making the logging interval come from a configurable property of `env`? And default it to 100?

`const loggingInterval = env.loggingInterval ?? 100`

File: lambdas/data-migration2/tests/test-executions.js
Comment (markdboyd) [inline lambdas/data-migration2/tests/test-executions.js:503]: If we make the logging interval configurable like I mentioned above, then we don't need to create 200 records for the unit test, which seems perhaps more ideal

File: lambdas/data-migration2/src/executions.ts
Comment (markdboyd) [reply lambdas/data-migration2/src/executions.ts:91]: If you agree I would make the change throughout the code

File: packages/db/src/types/summary.ts
Comment (markdboyd) [inline packages/db/src/types/summary.ts:1]: Nit: I would rename this file to `migration` for more clarity about the types

**Review by markdboyd — COMMENTED** [review]
This is great! I had a couple of optional comments for consideration

File: lambdas/data-migration2/src/executions.ts
Comment (botanical) [reply lambdas/data-migration2/src/executions.ts:91]: yeah, I think that's reasonable!

File: lambdas/data-migration2/src/executions.ts
Comment (botanical) [reply lambdas/data-migration2/src/executions.ts:24]: Oops, yeah it still throws it. Let me add that back

File: packages/db/src/types/migration.ts
Comment (Jkovarik) [inline packages/db/src/types/migration.ts:1]: I'm wondering if these more properly belong in the Core types package?     They're .... database migrations, but not technically *database* types.

File: lambdas/data-migration1/src/index.ts
Comment (Jkovarik) [inline lambdas/data-migration1/src/index.ts:52]: Interesting - so you don't need to JSON.parse(result) here?

**Review by Jkovarik — CHANGES_REQUESTED** [review]
I'd like to quickly discuss the db types comment, but this otherwise looks great, nice work!

File: lambdas/data-migration1/src/async-operations.ts
Comment (botanical) [reply lambdas/data-migration1/src/async-operations.ts:78]: My intention was to rely on the `skipped` but I can see how `RecordAlreadyMigrated <-> skipped` is not an automatic assumption users would make

File: lambdas/data-migration1/src/index.ts
Comment (botanical) [reply lambdas/data-migration1/src/index.ts:52]: That's a good point. I don't think it does. I'll fix that

File: lambdas/data-migration2/src/executions.ts
Comment (botanical) [reply lambdas/data-migration2/src/executions.ts:25]: Hm perhaps I should add this back in?

File: packages/db/src/types/migration.ts
Comment (botanical) [reply packages/db/src/types/migration.ts:1]: Hm I think this is definitely a good point. I'm inclined to leave it here because they are related to the migration even though they aren't technically database types

Comment (Jkovarik) [reply packages/db/src/types/migration.ts:1]: Not a hill I'm willing to die on.        If it weren't two migration scripts and/or we were keeping these around I'd be far more inclined to make a fuss about pulling it back in to the Lambda, but none of that needs to hold up the PR.      I'll consider the offline convo enough to unblock.

File: lambdas/data-migration1/src/index.ts
Comment (botanical) [reply lambdas/data-migration1/src/index.ts]: [`7d2cdeb` (#2173)](https://github.com/nasa/cumulus/pull/2173/commits/7d2cdeb41c225f4ed1270202abdc7a24e29b69de)

File: lambdas/data-migration2/src/granulesAndFiles.ts
Comment (botanical) [reply lambdas/data-migration2/src/granulesAndFiles.ts:189]: I think that's a good point! If this were going to be long lived, frequently used code, I would DRY it but for now, I think I will leave it as is.

File: lambdas/data-migration1/src/async-operations.ts
Comment (botanical) [reply lambdas/data-migration1/src/async-operations.ts:72]: [`a23d5fb` (#2173)](https://github.com/nasa/cumulus/pull/2173/commits/a23d5fb8244e2a8a4b8bfaca9814a26bc16680d1)

File: packages/db/src/types/migration.ts
Comment (botanical) [reply packages/db/src/types/migration.ts]: Okay, after the offline conversation, I have decided to move the types. I'm not sure if I should create a directory or if this is sufficient [`f81181c` (#2173)](https://github.com/nasa/cumulus/pull/2173/commits/f81181cae9208ce31628820c487010838e22dca3)

**Review by Jkovarik — APPROVED** [review]
This looks good, good work!

**Pull request merged**
